framework/src/actions/CreateExecutionerAction.C

```diff
@@ -38,8 +40,14 @@ CreateExecutionerAction::act()
   _moose_object_pars.set<FEProblemBase *>("_fe_problem_base") = _problem.get();
   _moose_object_pars.set<FEProblem *>("_fe_problem") =
       (std::dynamic_pointer_cast<FEProblem>(_problem)).get();
+  _moose_object_pars.set<EigenProblem *>("_eigen_problem") =
+      (std::dynamic_pointer_cast<EigenProblem>(_problem)).get();
```

Comment:
If succeed, I know this is an `eigenproblem`, otherwise it may be a regular `FEProblem`. I need this result to check if the `executioner` is consistent with `problem`.

I need `EigenExecutioner` work with `EigenProblem` only. 

Any other better way to get this goal?